pkg/client/unversioned/helper.go

```diff
@@ -361,10 +362,15 @@ func SetKubernetesDefaults(config *Config) error {
 	if len(config.UserAgent) == 0 {
 		config.UserAgent = DefaultKubernetesUserAgent()
 	}
-	if config.GroupVersion == nil {
-		config.GroupVersion = defaultVersionFor(config)
+	// TODO: Unconditionally set the config.Version, until we fix the config.
+	g, err := latest.Group("")
```

Comment:
This fixes the tests failure.

The tests failed because in this PR we fixed NegotiatedVersions(), it returns "extensions/v1beta1" if we try to create a resource that is "extensions/v1beta1", so the config here will have version="extensions/v1beta1". However,  `SetKubernetesDefaults` only recognizes "v1" at line 373 in this file, so it will fail.

A proper fix will require fix Config.Version, making that a `[]unversioned.GroupVersion`, but that's far beyond the scope of this PR.